src/core/reactor.cc

```diff
@@ -1946,7 +1946,7 @@ reactor::chmod(std::string_view name, file_permissions permissions) noexcept {
     });
 }
 
-directory_entry_type stat_to_entry_type(__mode_t type) {
+directory_entry_type stat_to_entry_type(mode_t type) {
```

Comment:
might want to add a prefix like

> reactor: replace __mode_t with mode_t

and explain the rationales in the body of commit message

> as __mode_t is glib specific, while mode_t is specified by POSIX, see https://man7.org/linux/man-pages/man3/mode_t.3type.html